cabal-testsuite/PackageTests/ConditionalAndImport/cabal-cyclical.project

```diff
@@ -0,0 +1,3 @@
+packages: .
+
+import: cabal-cyclical.project
```

Comment:
Maybe i am a little bit paranoic and probably the code already take it in account but maybe testing an indirect cycle would be good: cabal-root.project -> cabal-child.project -> cabal-root.project

Reply:
Also i miss a simple positive test over imports and conditionals, to exercise and demosntrate the pr main goal.

Reply:
https://github.com/haskell/cabal/pull/7783#issuecomment-1027334061

Reply:
> Also i miss a simple positive test over imports and conditionals, to exercise and demosntrate the pr main goal.

oops, the same test is testing both things, in addition to cycles, sorry, i was assuming they were to going placed in two separate tests (not needed)

Reply:
> [#7783 (comment)](https://github.com/haskell/cabal/pull/7783#issuecomment-1027334061)

afaics the test only checks a cabal.project cant import itself but no a cycle with more then one node (maybe it is not possible by design??)

Reply:
just checked it locally and the cycle check works perfectly, but i can imagine a bug breaking that invariant while not allowing auto imports

Reply:
and a test checking remote imports would not be bad as well, it could be in other pr as well

Reply:
i'm fine with a remote import test -- currently we have no tests relying on network connectivity so i didn't want to add one. but i think the cycle check test is fine as is -- indeed by design we check in a way that's robust to any cycle, so checking for 3-cycles vs 2-cycles isn't telling us anything special.

Reply:
> i'm fine with a remote import test -- currently we have no tests relying on network connectivity so i didn't want to add one. 

i've just checked 

```
packages: .
import: file://D:/ws/haskell/cabal/cabal.project
```

works. which is really nice. It would test part of the code path and will not go out the file system, no?